pennylane_forest/qvm.py

```diff
@@ -106,28 +106,7 @@ def __init__(self, device, *, shots=1000, noisy=False, **kwargs):
         if analytic:
             raise ValueError("QVM device cannot be run in analytic=True mode.")
 
-        # get the number of wires
-        if isinstance(device, nx.Graph):
-            # load a QVM based on a graph topology
-            num_wires = device.number_of_nodes()
-        elif isinstance(device, str):
-            # the device string must match a valid QVM device, i.e.
-            # N-qvm, or 9q-square-qvm, or Aspen-1-16Q-A
-            wire_match = re.search(r"(\d+)(q|Q)", device)
-
-            if wire_match is None:
-                # with the current Rigetti naming scheme, this error should never
-                # appear as long as the QVM quantum computer has the correct name
-                raise ValueError("QVM device string does not indicate the number of qubits!")
-
-            num_wires = int(wire_match.groups()[0])
-        else:
-            raise ValueError(
-                "Required argument device must be a string corresponding to "
-                "a valid QVM quantum computer, or a NetworkX graph object."
-            )
-
-        super().__init__(num_wires, shots, analytic=analytic, **kwargs)
```

Comment:
Now that we load the number of wires in a different way, I think all this block does is input validation of `device`. However, this is something that is also covered by `get_qc` so I don't think we need to repeat here.

Reply:
Agree!

Reply:
This actually simplifies the plugin logic significantly